hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/JobMetrics.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.internal.metrics.MetricsUtil;
+import com.hazelcast.spi.annotation.Beta;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.Util.entry;
+
+/**
+ * An immutable snapshot of job-specific metrics.
+ */
+@Beta
+public final class JobMetrics implements Serializable {
```

Comment:
we should implement custom serialization (i.e. IdentifiedDataSerializable) for this class